sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/link.py

```diff
@@ -239,11 +239,11 @@ def _incoming_detach(self, frame) -> None:
             self._error = error_cls(condition=frame[2][0], description=frame[2][1], info=frame[2][2])
             self._set_state(LinkState.ERROR)
         else:
-            if self.state != LinkState.DETACH_SENT:
+            # if self.state != LinkState.DETACH_SENT:
                 # Handle the case of when the remote side detaches without sending an error.
                 # We should detach as per the spec but then retry connecting
-                self._error = AMQPLinkError(condition=ErrorCondition.UnknownError,
-                                          description="Link detached unexpectedly.", retryable=True)
+            self._error = AMQPLinkError(condition=ErrorCondition.UnknownError,
+                                        description="Link detached unexpectedly.", retryable=True)
```

Comment:
we might want to change the error description based on whether a detach was just sent or not + update comment to include the detach sent first scenario